app/src/lib/databases/repositories-database.ts

```diff
@@ -59,6 +59,8 @@ export interface IDatabaseRepository {
    * of Git and GitHub.
    */
   readonly isTutorialRepository?: boolean
+
+  readonly codespace?: boolean
```

Comment:
this should probably be more abstract, for 2 reasons:

1. This isn't _really_ about Codespaces, it's about remote execution.
2. A boolean only gives us support for a single codespace at a time w/o resorting to some sort of registration system to bookkeep the networking requirements.